recipe/recipe.yaml

```diff
@@ -0,0 +1,577 @@
+context:
+  version: "7.8.3"
+
+recipe:
+  name: suitesparse
+  version: ${{ version }}
+
+source:
+  url: https://github.com/DrTimothyAldenDavis/SuiteSparse/archive/v${{ version }}.tar.gz
+  sha256: ce39b28d4038a09c14f21e02c664401be73c0cb96a9198418d6a98a7db73a259
+
+build:
+  number: 2
+  string: ss${{ version | replace('.', '') }}_h${{ hash }}
+
+outputs:
+  - package:
+      name: libsuitesparseconfig
+    about:
+      license: BSD-3-Clause
+      license_file: LICENSE.txt
+    build:
+      script:
+        # rattler-build adds `.sh` or `.bat` based on platform
+        file: build-one
+        env:
+          SUBPKG_DIR: SuiteSparse_config
+    requirements:
+      build:
+        - ${{ compiler('c') }}
+        - ${{ stdlib('c') }}
+        - ${{ compiler('fortran') }}
+        - cmake
+        - ninja
+        - pkg-config
+      host:
+        - if: osx
+          then:
+            - llvm-openmp
+        - if: linux
+          then:
+            - libgomp
+        - libblas
+        - libcblas
+        - liblapack
+      ignore_run_exports:
+        from_package:
+          - libblas
+          - libcblas
+          - liblapack
+          - ${{ fortran_compiler }}
+      run_exports:
+        - ${{ pin_subpackage('libsuitesparseconfig') }}
+    tests:
+      - package_contents:
+          lib:
+            - suitesparseconfig
+  - package:
+      name: libamd
+      version: 3.3.3
```

Comment:
Could we automate how the package-specific versions get populated? Or at least auto-validate it based on some in-repo information?

I feel it'll be tricky to update multiple versions correctly if everything is done manually (and the bot will not do it either, leading to an even more tempting "just merge a green PR" case where this could happen)

Reply:
> Could we automate how the package-specific versions get populated?

I think it's impossible to automate _populating_ the versions without implementing special support in conda-smithy.

> Or at least auto-validate it based on some in-repo information?

Yes, it is validated in build.sh, specifically to avoid these accidental incorrect versions (you're right, it is technically SOVERSION). suitesparse uses default cmake behavior, where SOVERSION is set from package VERSION, so these are necessarily the same (it's a lot more complicated and more likely to differ with autotools). That technically _could_ change, so I could update the check to extract the cmake package version directly, if you prefer. I'll look into that.